helper/schema/grpc_provider_test.go

```diff
@@ -2319,7 +2319,13 @@ func TestStopContext_stop(t *testing.T) {
 				}
 				close(doneCh)
 			}()
-			server.StopProvider(context.Background(), &tfprotov5.StopProviderRequest{})
+
+			_, err = server.StopProvider(context.Background(), &tfprotov5.StopProviderRequest{})
+
+			if err != nil {
+				t.Fatalf("unexpected StopProvider error: %s", err)
+			}
```

Comment:
Why are you not sticking with the _if with assignment_ that is used everywhere else above in this PR?

Looking at the rest of the changes for this file, seem like it's deliberate, so I'm curious.

Reply:
Great question! Usually its objectively easier to read split assignment from error checking in larger functions for [line of sight in code](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88), but in this case it can probably be consolidated since we don't need the `err` for other reasons.

I'm not sure if we have any particular stylistic choice for this in general, but if we did want one, there's probably some static analysis tooling which can enforce it. That's generally my requirement for stylistic things -- it must be enforceable via tooling, rather than trying to manually enforce via human reviews. 👍

Reply:
Yeah, I completely agree that code-style needs to be tool-enforced. It really matters little (and Go has a fairly lean grammar anyway), but this way the tool sets a style and engineers can focus on other stuff.

I'll give that article a read now.

FYI my question was frankly spawned out of "why is he doing this one file differently than the rest of this changeset", it's not like I was try to suggest a specific style. I'm much more for consistency than for a specific "rule".